lib/on_message.js

```diff
@@ -0,0 +1,14 @@
+var on_message = function(haxfred, message) {
```

Comment:
(total nitpick)
Is there a reason you are assigning the function to var instead of just using a named function and exporting that?